### DISCUSSION TITLE
Is React 19 & R3F v9 Supported? Or am I Doing It Wrong™

### DISCUSSION THREAD

**Discussion by sxxov** [pmndrs/drei#2213] [General]
hi! i've been eyeing drei since i've started using three.js, & i've finally gone through the process of getting up to speed with react & r3f to be able to take advantage of this project. since i'm starting from scratch, i went with the decision of targetting react 19 after finding out r3f has betas & RC's out to support the new react version.

however, after plugging in various drei components like `MeshPortalMaterial` & `RenderTexture` (as well as `EffectComposer`, but i'm not sure if this is the correct place to ask), they required manual patches to their usages of `__r3f` as they tried to access undefined properties (e.g. `object.parent` is assumed to be `Object3D` with another `__r3f` property, but is actually an object with a `object` property that points to the `Object3D`).

my question is, am i configuring/doing something wrong? or are the utilities here not compatible with my setup yet? the following are some package versions that might be of interest:

```
"@react-three/drei": "^9.116.2",
"@react-three/fiber": "^9.0.0-rc.1",
"@react-three/postprocessing": "^2.16.3",
"next": "^15.0.3",
"postprocessing": "^6.36.4",
"react": "^19.0.0-rc-603e6108-20241029",
"react-dom": "^19.0.0-rc-603e6108-20241029",
"three": "^0.170.0",
"three-stdlib": "^2.34.0",
```

**Comment by talupoeg** [comment]
https://github.com/pmndrs/drei/releases/tag/v10.0.0-rc.0
